src/meta/src/hummock/manager/compaction_group_manager.rs

```diff
@@ -337,44 +368,48 @@ impl HummockManager {
                 delta_type: Some(DeltaType::GroupDestroy(GroupDestroy {})),
             });
         }
-        new_version_delta.pre_apply();
-        commit_multi_var!(self.meta_store_ref(), version)?;
 
         for (group_id, max_level) in groups_to_remove {
             remove_compaction_group_in_sst_stat(&self.metrics, group_id, max_level);
         }
 
+        new_version_delta.pre_apply();
+
         // Purge may cause write to meta store. If it hurts performance while holding versioning
         // lock, consider to make it in batch.
-        self.compaction_group_manager
-            .write()
-            .await
-            .purge(HashSet::from_iter(get_compaction_group_ids(
-                &versioning.current_version,
-            )))
-            .await?;
+        let mut compaction_group_manager = self.compaction_group_manager.write().await;
+        let mut compaction_groups_txn = compaction_group_manager.start_compaction_groups_txn();
+
+        if compaction_groups_txn.purge(HashSet::from_iter(get_compaction_group_ids(
+            version.latest_version(),
+        ))) {
+            commit_multi_var!(self.meta_store_ref(), version, compaction_groups_txn)?;
+        } else {
+            commit_multi_var!(self.meta_store_ref(), version)?;
+        }
+
         Ok(())
     }
 
```

Comment:
`try_update_write_limits` will aquire the compaction_group_manager write lock while the write lock is held here. Will this be an issue.